### PR TITLE
[TextFields] Implementing textFieldShouldClear:.

### LINKED ISSUE #2062: [TextFields] textFieldShouldClear is not invoked when clear button is tapped
The `UITextFieldDelegate` method `textFieldShouldClear` should be invoked when the clear button is tapped on an MDCTextField.  

https://developer.apple.com/documentation/uikit/uitextfielddelegate/1619594-textfieldshouldclear

### PR DESCRIPTION
**Pull request by willlarche**
Closes #2062 

Also added an equivalent to MDCMultilineTextField.

### CODE REVIEW

File: components/TextFields/src/MDCMultilineTextField.h [hunk 1]
```diff
@@ -29,6 +30,9 @@
 /** A mirror of the same property that already exists on UITextField, UITextView, and UILabel. */
 @property(nonatomic, assign) BOOL adjustsFontForContentSizeCategory;
 
+/** An optional delegate for useful methods not included in UITextViewDelegate.*/
+@property(nonatomic, nullable, weak) IBOutlet id<MDCMultilineTextInputDelegate> multilineDelegate;
```
Comment (romoore) [inline components/TextFields/src/MDCMultilineTextField.h:34]: Follow-up PR: Unit tests for encoding/decoding?

**Review by romoore — APPROVED** [review]

File: components/TextFields/src/MDCMultilineTextField.h (hunk 1 above)
Comment (willlarche) [reply components/TextFields/src/MDCMultilineTextField.h:34]: I couldn’t figure out how. I don’t have any classes that conform.

Comment (romoore) [reply components/TextFields/src/MDCMultilineTextField.h:34]: You could create a custom test class that conforms and encode/decode. If you wanted to be super-sneaky, slap those both into an NSArray, encode/decode the array and verify that the relationship still holds.

Comment (willlarche) [reply components/TextFields/src/MDCMultilineTextField.h:34]: Do I need to make that class in another file or can I do it in the unit test file at the top?

Comment (romoore) [reply components/TextFields/src/MDCMultilineTextField.h:34]: I put them at the top of the test file until I have more than one. Then it's a test Supplemental?

**Pull request merged**